web/client/src/routes/+page.svelte

```diff
@@ -296,6 +297,35 @@
     });
   }
 
+  function reportExecutionFinished() {
```

Comment:
This function looks the same as `reportExists` with only a URL as a difference. We can reuse `reportExists` by renaming it with a clear name, something like _remoteFileExists_, and adding a URL parameter.